src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let separ = 'null';
+
+    if (separator !== null) {
+      separ = separator.toString();
+    }
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let value = this[i];
+
+      if (value === undefined || value === null) {
+        value = '';
+      }
+
+      result += value;
```

Comment:
Here it is better to check for not equal and immediately add the result inside. 
Try to rewrite this